WalletWasabi/Hwi2/HwiClient.cs

```diff
@@ -0,0 +1,305 @@
+using NBitcoin;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using WalletWasabi.Hwi2.Exceptions;
+using WalletWasabi.Hwi2.Models;
+using WalletWasabi.Hwi2.Parsers;
+using WalletWasabi.Hwi2.ProcessBridge;
+using WalletWasabi.Interfaces;
+using WalletWasabi.KeyManagement;
+
+namespace WalletWasabi.Hwi2
+{
+	public class HwiClient
+	{
+		#region PropertiesAndMembers
+
+		public Network Network { get; }
+		public IProcessBridge Bridge { get; }
+
+		#endregion PropertiesAndMembers
+
+		#region ConstructorsAndInitializers
+
+		public HwiClient(Network network, IProcessBridge bridge = null)
+		{
+			Network = Guard.NotNull(nameof(network), network);
+			Bridge = bridge ?? new HwiProcessBridge();
+		}
+
+		#endregion ConstructorsAndInitializers
+
+		#region Commands
+
+		private async Task<string> SendCommandAsync(IEnumerable<HwiOption> options, HwiCommands? command, string commandArguments, bool openConsole, CancellationToken cancel)
+		{
+			string arguments = HwiParser.ToArgumentString(Network, options, command, commandArguments);
+
+			try
+			{
+				(string responseString, int exitCode) = await Bridge.SendCommandAsync(arguments, openConsole, cancel).ConfigureAwait(false);
+
+				if (exitCode != 0)
+				{
+					ThrowIfError(responseString);
```

Comment:
This could be extracted out of the `if` statement and also remove the one after the `if` statement block because it will be redundant.

Reply:
exitCode and responseString are not attached. Sometimes the error indicated by exitCode, sometimes by resposeString, sometimes both.